File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int[] maxSize = new int[sizes.length];
        int[] minSize = new int[sizes.length];
        
        for (int i = 0; i < sizes.length; i++) {
            if (sizes[i][0] >= sizes[i][1]) {
                maxSize[i] = sizes[i][0];
                minSize[i] = sizes[i][1];
            }
            else {
                maxSize[i] = sizes[i][1];
                minSize[i] = sizes[i][0];
            }
        }
        
        Arrays.sort(maxSize);
        Arrays.sort(minSize);
    
        return maxSize[sizes.length-1] * minSize[sizes.length-1];
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
     
    int max_row = 0;
    int max_col = 0;
    
    for (vector<int>& size : sizes) {
        if (size[0] > size[1] ) {
            if (max_row < size[0]) {
                max_row = size[0];
            }
            if (max_col < size[1]) {
                max_col = size[1];
            }
        }
        else {
            if (max_row < size[1]) {
                max_row = size[1];
            }
            if (max_col < size[0]) {
                max_col = size[0];
            }
        }
    }
    
    return max_row * max_col;
}